torchserve/README.md

```diff
@@ -0,0 +1,248 @@
+
+For instance:
+
+```
+metrics_address=http://0.0.0.0:8082
+metrics_mode=prometheus
+```
+
+In this case, the OpenMetrics endpoint will be exposed at this URL: `http://<TORCHSERVE_ADDRESS>:8082/metrics`.
+
+### Configuration
+
+These three different endpoints can be monitored independently and must be configured separately in the configuration file, one API per instance. See the [sample torchserve.d/conf.yaml][4] for all available configuration options.
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "OpenMetrics endpoint" xxx -->
+#### Configure the OpenMetrics endpoint
+
+Configuration options for the OpenMetrics endpoint can be found in the configuration file under the `TorchServe OpenMetrics endpoint configuration` section. The minial configuration only requires the `openmetrics_endpoint` option:
+
+```yaml
+init_config:
+  ...
+instances:
+  - openmetrics_endpoint: http://<TORCHSERVE_ADDRESS>:8082/metrics
+```
+
+A lot more options are available for this endpoint, as you can see in the [sample torchserve.d/conf.yaml][4].
+
+TorchServe allows the custom service code to emit metrics that will be available based on the configured `metrics_mode` as explained in [their documentation][11]. You can configure this integration to collect these metrics using the `extra_metrics` option. These metrics will have the `torchserver.openmetrics` prefix, just like any other metrics coming from this endpoint.
+
+<div class="alert alert-info">These custom TorchServe metrics will be considered as standard metrics in Datadog.</div>
+
+<!-- xxz tab xxx -->
+<!-- xxx tab "Inference API" xxx -->
+#### Configure the Inference API
+
+This integration relies on the [Inference API][14] to get the overall status of your TorchServe instance. Configuration options for the Inference API can be found in the [configuration file][4] under the `TorchServe Inference API endpoint configuration` section. The minial configuration only requires the `inference_api_url` option:
+
+```yaml
+init_config:
+  ...
+instances:
+  - inference_api_url: http://<TORCHSERVE_ADDRESS>:8080
+```
+
+Under the hood, this integration leverages the [ping][12] endpoint to collect the overall health status of your TorchServe server.
+
+<!-- xxz tab xxx -->
+<!-- xxx tab "Management API" xxx -->
+#### Configure the Management API
+
+You can collect metrics related to the models that are currently running in your TorchServe server using the [Management API][13]. Configuration options for the Inference API can be found in the [configuration file][4] under the `TorchServe Management API endpoint configuration` section. The minial configuration only requires the `management_api_url` option:
+
+```yaml
+init_config:
+  ...
+instances:
+  - management_api_url: http://<TORCHSERVE_ADDRESS>:8081
+```
+
+By default, the integration collects data from every single models, up to 100 models. This can be modified using the `limit`, `include` and `exclude` options. For example:
```

Comment:
```suggestion
By default, the integration collects data from every single models, up to 100 models. This can be modified using the `limit`, `include`, and `exclude` options. For example:
```